torch/csrc/jit/script/compiler.cpp

```diff
@@ -1499,11 +1553,27 @@ struct to_ir {
 
   std::shared_ptr<SugaredValue> emitApplyExpr(Apply &apply, size_t n_binders) {
     auto sv = emitSugaredExpr(apply.callee(), 1);
-    auto inputs = getNamedValues(apply.inputs(), true);
     auto attributes = fmap(apply.attributes(), [&](const Attribute& attr) {
       return NamedValue(attr.range(), attr.name().name(), emitExpr(attr.value()));
     });
-    return sv->call(apply.callee().range(), method, inputs, attributes, n_binders);
+
+    auto loc = apply.callee().range();
+    if (sv->kind() == "fork") {
+      JIT_ASSERT(attributes.size() == 0);
```

Comment:
You have to throw a user interpretable error for every user-hittable thing. Asserts are only appropriate for errors that indicate a bug in the compiler.
```torch.jit.fork(foo, a=5)` will trigger attributes.```